src/pydata_sphinx_theme/theme/pydata_sphinx_theme/sections/sidebar-primary.html

```diff
@@ -1,14 +1,8 @@
 {% block docs_sidebar %}
 {# Check whether there are sidebar links to conditionally hide the sidebar #}
 {# copied over from ../components/sidebar-nav-bs.html #}
-{% set sidebar_nav_links = generate_nav_html("sidebar",
-                              show_nav_level=theme_show_nav_level|int,
-                              maxdepth=theme_navigation_depth|int,
-                              collapse=theme_collapse_navigation|tobool,
-                              includehidden=True,
-                              titles_only=True) %}
 
-{% if sidebars and sidebar_nav_links %}
+{% if sidebars %}
```

Comment:
Something we could do to simplify this further and have similar behavior:

In `layout.html`, add a line like:

```
{# Create the sidebar links HTML here to re-use in a few places #}
{# If we have no sidebar links, pop the links component from the sidebar list #}
{%- set sidebar_nav_html = generate_nav_html("sidebar",
    show_nav_level=theme_show_nav_level|int,
    maxdepth=theme_navigation_depth|int,
    collapse=theme_collapse_navigation|tobool,
    includehidden=True,
    titles_only=True)
-%}
{% if not sidebar_nav_html %}
{% set sidebars = sidebars | reject("in", "sidebar-nav-bs.html") | list %}
{% endif %}
```

Then we can just re-use `sidebar_nav_html` in our `sidebar-primary` template (`{{ sidebar_nav_html }}`). AND, if that HTML is empty, then we remove the `sidebar-nav-bs` from our `sidebars` list. That list will then be empty if that was the only template in it, and will then be hidden.